peach/src/main/java/com/hartwig/hmftools/peach/haplotype/HaplotypeCombination.java

```diff
@@ -0,0 +1,68 @@
+package com.hartwig.hmftools.peach.haplotype;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hartwig.hmftools.peach.PeachUtils.GERMLINE_TOTAL_COPY_NUMBER;
+
+public class HaplotypeCombination
+{
+    @NotNull
+    private final Map<String, Integer> haplotypeNameToCount;
+
+    public HaplotypeCombination(@NotNull Map<String, Integer> haplotypeNameToCount)
+    {
+        if (haplotypeNameToCount.values().stream().anyMatch(c -> c <= 0))
+        {
+            String error_msg = String.format(
+                    "Invalid haplotype combination with non-positive count(s): %s", haplotypeNameToCount
```

Comment:
How can counts become negative?

Reply:
They shouldn't be able to. I would only be possible if some kind of bug was at play. I just put this in here to make sure such bugs would be caught before creating output files.

Would it be better to skip such tests/checks?